shenyu-registry/shenyu-registry-polaris/src/main/java/org/apache/shenyu/registry/polaris/PolarisInstanceRegisterRepository.java

```diff
@@ -0,0 +1,144 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shenyu.registry.polaris;
+
+import com.tencent.polaris.api.config.Configuration;
+import com.tencent.polaris.api.core.ConsumerAPI;
+import com.tencent.polaris.api.core.ProviderAPI;
+import com.tencent.polaris.api.exception.PolarisException;
+import com.tencent.polaris.api.pojo.Instance;
+import com.tencent.polaris.api.rpc.GetHealthyInstancesRequest;
+import com.tencent.polaris.api.rpc.InstanceRegisterRequest;
+import com.tencent.polaris.api.rpc.InstancesResponse;
+import com.tencent.polaris.client.api.SDKContext;
+import com.tencent.polaris.factory.ConfigAPIFactory;
+import com.tencent.polaris.factory.api.DiscoveryAPIFactory;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.shenyu.common.constant.Constants;
+import org.apache.shenyu.common.constant.PolarisPathConstants;
+import org.apache.shenyu.common.exception.ShenyuException;
+import org.apache.shenyu.registry.api.ShenyuInstanceRegisterRepository;
+import org.apache.shenyu.registry.api.config.RegisterConfig;
+import org.apache.shenyu.registry.api.entity.InstanceEntity;
+import org.apache.shenyu.spi.Join;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The type polaris instance register repository.
+ */
+@Join
+public class PolarisInstanceRegisterRepository implements ShenyuInstanceRegisterRepository {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PolarisInstanceRegisterRepository.class);
+
+    private ConsumerAPI consumerAPI;
+
+    private ProviderAPI providerAPI;
+
+    private String namespace;
+
+    /**
+     * Init.
+     * @param config the config
+     */
+    @Override
+    public void init(final RegisterConfig config) {
+        Configuration configuration = buildConfiguration(config);
+        SDKContext sdkContext = SDKContext.initContextByConfig(configuration);
+
+        this.consumerAPI = DiscoveryAPIFactory.createConsumerAPIByContext(sdkContext);
+        this.providerAPI = DiscoveryAPIFactory.createProviderAPIByContext(sdkContext);
+        this.namespace = config.getProps().getProperty(Constants.NAMESPACE, PolarisPathConstants.NAMESPACE);
+    }
+
+    private Configuration buildConfiguration(final RegisterConfig config) {
+        String serverLists = config.getServerLists();
+        return ConfigAPIFactory.createConfigurationByAddress(serverLists);
+    }
+
+    /**
+     * Persist instance.
+     * @param instance instance
+     */
+    @Override
+    public void persistInstance(final InstanceEntity instance) {
+        final InstanceRegisterRequest req = new InstanceRegisterRequest();
+        req.setInstanceId(buildInstanceNodeName(instance));
+        req.setWeight(1);
```

Comment:
why 1?

Reply:
refer to 'org.apache.shenyu.registry.nacos.NacosInstanceRegisterRepository#persistInstance';
`
    org.apache.shenyu.registry.nacos.NacosInstanceRegisterRepository#persistInstance

    @Override
    public void persistInstance(final InstanceEntity instance) {
        try {
            Instance inst = new Instance();
            inst.setWeight(1.0d);
            inst.setEphemeral(true);
            inst.setIp(instance.getHost());
            inst.setPort(instance.getPort());
            inst.setInstanceId(buildInstanceNodeName(instance));
            inst.setServiceName(instance.getAppName());
            namingService.registerInstance(instance.getAppName(), groupName, inst);
            LOGGER.info("nacos client register success: {}", inst);
        } catch (NacosException e) {
            throw new ShenyuException(e);
        }
    }
`